osquery/core/init.cpp

```diff
@@ -97,13 +101,17 @@ enum {
 namespace {
 extern "C" {
 static inline bool hasWorkerVariable() {
-  return (getenv("OSQUERY_WORKER") != nullptr);
+  if (auto value = ::osquery::getEnvVar("OSQUERY_WORKER")) {
```

Comment:
This is confusing, I know how it works ;), but it doesn't read well. 

"If I can assign a variable, this is true!?" Mind making this simpler? Grab some stack, and have the conditional check the resultant value of your data.